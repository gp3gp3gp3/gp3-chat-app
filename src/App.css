.App {
  text-align: center;
  background-color: #fafbfc;
}

.Body {
  margin: 0 30%;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  background-color: white;
  min-height: calc(100vh - 100px);
}
