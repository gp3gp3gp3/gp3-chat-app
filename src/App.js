import React, { Component } from 'react'
import ReactDOM from 'react-dom'
import firebase from 'firebase'
import Header from './components/Header/Header'
import MessageList from './containers/MessageList'
import MessageBox from './containers/MessageBox'
import './App.css'

class App extends Component {
  constructor () {
    super()
    const config = {
      apiKey: 'AIzaSyBHymkx-sx64eGJFLNmv8Y7nwakdXCNcpQ',
      authDomain: 'gp3-chat-app.firebaseapp.com',
      databaseURL: 'https://gp3-chat-app.firebaseio.com',
      projectId: 'gp3-chat-app',
      storageBucket: '',
      messagingSenderId: '111539696639'
    }
    firebase.initializeApp(config)
    this.scrollToBottom = this.scrollToBottom.bind(this)
  }

  scrollToBottom () {
    const node = ReactDOM.findDOMNode(this.refs['messageBox'])
    window.scrollTo({
      left: 0,
      top: node.offsetTop,
      behavior: 'smooth'
    })
  }

  render () {
    return (
      <div className='App'>
        <Header />
        <div className='Body'>
          <MessageList
            db={firebase}
            scrollToBottom={this.scrollToBottom}
          />
          <MessageBox
            ref='messageBox'
            db={firebase}
          />
        </div>
      </div>
    )
  }
}

export default App
