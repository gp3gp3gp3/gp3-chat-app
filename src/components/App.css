.App {
  text-align: center;
  background-color: #fafbfc;
}

.Body {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  background-color: white;
  min-height: calc(100vh - 100px);
}

@media only screen and (min-width: 450px) {
  .Body {
    margin: 0 5%;
  }
}

@media only screen and (min-width: 550px) {
  .Body {
    margin: 0 10%;
  }
}

@media only screen and (min-width: 700px) {
  .Body {
    margin: 0 15%;
  }
}

@media only screen and (min-width: 850px) {
  .Body {
    margin: 0 25%;
  }
}

@media only screen and (min-width: 1000px) {
  .Body {
    margin: 0 30%;
  }
}
