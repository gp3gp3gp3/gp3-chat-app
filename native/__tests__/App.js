/* eslint-env jest */
import 'react-native'
// Note: test renderer must be required after react-native.
import { shallow } from 'enzyme'
import React from 'react'
import App from '../src/App'

it('renders correctly', () => {
  const wrapper = shallow(<App />)
  expect(wrapper).toMatchSnapshot()
})
