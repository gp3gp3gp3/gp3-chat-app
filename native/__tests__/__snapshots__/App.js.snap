// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 20,
              "margin": 10,
              "textAlign": "center",
            }
          }
        >
          Welcome to React Native!
        </Text>,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#333333",
              "marginBottom": 5,
              "textAlign": "center",
            }
          }
        >
          To get started, edit App.js
        </Text>,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#333333",
              "marginBottom": 5,
              "textAlign": "center",
            }
          }
        >
          Press Cmd+R to reload,
Cmd+D or shake for dev menu
        </Text>,
      ],
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#F5FCFF",
        "flex": 1,
        "justifyContent": "center",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Welcome to React Native!",
          "ellipsizeMode": "tail",
          "style": Object {
            "fontSize": 20,
            "margin": 10,
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": "Welcome to React Native!",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "To get started, edit App.js",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "#333333",
            "marginBottom": 5,
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": "To get started, edit App.js",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Press Cmd+R to reload,
Cmd+D or shake for dev menu",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "#333333",
            "marginBottom": 5,
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": "Press Cmd+R to reload,
Cmd+D or shake for dev menu",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 20,
                "margin": 10,
                "textAlign": "center",
              }
            }
          >
            Welcome to React Native!
          </Text>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#333333",
                "marginBottom": 5,
                "textAlign": "center",
              }
            }
          >
            To get started, edit App.js
          </Text>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#333333",
                "marginBottom": 5,
                "textAlign": "center",
              }
            }
          >
            Press Cmd+R to reload,
Cmd+D or shake for dev menu
          </Text>,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#F5FCFF",
          "flex": 1,
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Welcome to React Native!",
            "ellipsizeMode": "tail",
            "style": Object {
              "fontSize": 20,
              "margin": 10,
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": "Welcome to React Native!",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "To get started, edit App.js",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#333333",
              "marginBottom": 5,
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": "To get started, edit App.js",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Press Cmd+R to reload,
Cmd+D or shake for dev menu",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#333333",
              "marginBottom": 5,
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": "Press Cmd+R to reload,
Cmd+D or shake for dev menu",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
